langgraph/pregel/__init__.py

```diff
@@ -546,8 +546,19 @@ def update_state(
             # apply to checkpoint and save
             _apply_writes(checkpoint, channels, task.writes)
             step = saved.metadata.get("step", -2) + 1 if saved else -1
+
+            # merge configurable fields with previous checkpoint config
+            checkpoint_config = config
```

Comment:
Double check: Are we supposed to update `update_state()` and `aupdate_state()`?

Reply:
Yes always update both

Reply:
Confirmed, we need this. When a tool query is manually updated, `update_state()` or `aupdate_state()` is called. The client needs to pass the same configurable fields for all thread update actions.